event_model/__init__.py

```diff
@@ -648,6 +699,10 @@ class EventModelRuntimeError(EventModelError, RuntimeError):
     ...
 
 
+class EventModelTypeError(EventModelError, TypeError):
+    ...
```

Comment:
Do you mind to add a docstring here as well to have a better understanding of the purpose of this exception?

Reply:
I think it's covered by 3ea1e63.

Reply:
Good point. If you look above and below you'll see this is one of many similar exceptions. I've added a comment in 3ea1e63b167cc0bc1cc8f6e54d79e153420449e7 to explain the motivation for all.

Reply:
OK, sounds reasonable.